### DISCUSSION TITLE
Add OpMode from Android Studio Mac (Solved)

### DISCUSSION THREAD

**Discussion by rhunter-NTatC** [FIRST-Tech-Challenge/FtcRobotController#212] [Q&A] [answered]
So to place an OpMode on the Driver Hub we're supposed to plug everything in, hit the green play button in Android Studio and choose the device.
Per Instructions: [https://github.com/FIRST-Tech-Challenge/FtcRobotController/wiki/Creating-and-Running-an-Op-Mode-(Android-Studio)](https://github.com/FIRST-Tech-Challenge/FtcRobotController/wiki/Creating-and-Running-an-Op-Mode-(Android-Studio) )

However, on a Mac there's no build configuration for the project.  The play button is greyed out.
Does anyone know how to correctly add a configuration for this?

Or is it just not possible with a Mac?

**Comment by texasdiaz** [comment]
Sorry, I don't mean to be pedantic but I want to make sure we're all talking the same language here. Are you trying to deploy to a Driver Hub, or to a Control Hub that is currently being powered by a robot battery?

**Reply by rhunter-NTatC** [reply]
Trying to deploy to a Driver Hub connected via usb. 

From everywhere I’ve looked that where the OpMode files need to go. But it’s very possible I’m reading them wrong.

**Reply by AlecHub** [reply]
OpMode gets loaded onto RobotController device.  More accurately,  RobotController app gets loaded onto RobotController device.  

Android Studio builds your OpMode into RobotController app and then loads the app unto RobotController device.  

l made the same mistake of thinking OpModes go on Driver Station when I first started out in FTC, not knowing that Driver Station is more or less a "dumb terminal".

**Reply by Windwoes** [reply]
To continue what Alec said, the RobotController sends a list of the available OpModes to the Driver Station, which is how the OpMode drop-down gets populated.

**Comment by rhunter-NTatC** [comment]
So, I must have gotten that into my head and read it into everything I saw after.  Thank-you!

However, there's still no configuration available.  
I'm wondering if the Mac won't recognize the Control Hub for some reason.
This is What I see at the top.  (Control Hub connected via orange USB-C cable) 
<img width="509" alt="Screen Shot 2021-10-22 at 8 56 14 AM" src="https://user-images.githubusercontent.com/72524582/138466458-25be583c-77dd-4007-8193-001d771056f3.png">

**Reply by rhunter-NTatC** [reply]
I got hold of a PC and installed everything.
I'm having the same exact problem.

**Reply by AlecHub** [reply]
As a sanity check, are you able to wirelessly "pair" a Driver Station to the Control Hub?  Also, try diagnosing the Control Hub via the [REV Hardware Client](https://docs.revrobotics.com/rev-hardware-client) App.

**Reply by Windwoes** [reply]
Are you sure you're opening the SDK project correctly in Android Studio?

**Reply by rhunter-NTatC** [reply]
(sI think I found the problem with the Mac.  There's an SDK licensing issue that I cant resolve until the district's tech support allows Android Studio to update itself.  I don't have enough managerial rights to get around it (without being fired for "hacking").  It's also not recognizing the Control Hub.

The PC is another issue(s).
It does recognize the Control Hub.  It shows up in Windows File Explorer as an external drive.

Also, if I choose "build" -> "Rebuild Project" I get "Unable to find Gradle tasks to build: [:]" in the event Log
![image](https://user-images.githubusercontent.com/72524582/138512799-e34e7970-55bb-4aa9-930c-a2d6e73c1fdc.png)
I'm wondering if there's a Gradle task I need to run first. (I've barely used Gradle)

**Reply by rhunter-NTatC** [reply]
Can I just drag the .java files into a specific folder in Windows Explorer?
That would be at least a temporary fix.
![image](https://user-images.githubusercontent.com/72524582/138513910-a6869624-cf8b-4085-9a96-9557431596d1.png)

**Comment by cmacfarl** [comment]
Screenshots are very welcome when asking these sorts of questions.

The "Loading Devices..." text in the devices dropdown seems to indicate that your Mac can't see the Control Hub.  

If you access to a PC, does the REV Hardware Client see the Control Hub when you follow the directions here?

https://docs.revrobotics.com/rev-control-system/control-hub-gs/connect-to-the-control-hub-robot-control-console

**Reply by rhunter-NTatC** [reply]
No it doesn't.  See the response from about 3:30 (didn't want to post twice)

**Comment by rhunter-NTatC** [comment, marked as answer]
FIXED!!!

Thank you all!

I found this video on another forum showing how to set up ArcticFox appropriately.
[https://youtu.be/aFvmt47xHNY](https://youtu.be/aFvmt47xHNY)

**Reply by AlecHub** [reply]
> FIXED!!!
> 
> Thank you all!
> 
> I found this video on another forum showing how to set up ArcticFox appropriately. https://youtu.be/aFvmt47xHNY

Good stuff!  Is there a "**Mark as answer**" button on your post?  If so, you should click on it.

**Reply by rhunter-NTatC** [reply]
Just did.
Thank you for the reminder.
